service/resource/legacy/daemonset.go

```diff
@@ -88,6 +89,20 @@ func newDaemonSetContainers(spec flanneltpr.Spec, etcdCAFile, etcdCrtFile, etcdK
 					},
 				},
 			},
+			LivenessProbe: &api.Probe{
+				InitialDelaySeconds: initialDelaySeconds,
+				TimeoutSeconds:      timeoutSeconds,
+				PeriodSeconds:       periodSeconds,
+				FailureThreshold:    failureThreshold,
+				SuccessThreshold:    successThreshold,
+				Handler: api.Handler{
+					HTTPGet: &api.HTTPGetAction{
+						Path: healthEndpoint,
+						Port: intstr.IntOrString{IntVal: livenessPort(spec)},
+						Host: probeHost,
+					},
+				},
+			},
```

Comment:
This is a bit new to me. I understand it as we configure a health check by looking at the state of a different container. In case the other container fails the health check fails and restarts the pod? Or does it restart the container? I am wondering why we need this redirect. I thought we just define a health check for the new container and in case this fails the pod restarts with all its containers. So there would be no need to do this redirect and define the health checks everywhere.

Reply:
OK, you got me there,  I was thinking that if health check fails it will reboot only the container. I need to check this

Reply:
I did some tests and the restart of `container-based` not `pod-based`.
If container fail health-check only that container is restarted rest if left untouched.
```
calvix@0000dd1bb02a188c ~ $ docker ps | grep flannel-network-sml8f
547942c3f5a4        quay.io/coreos/flannel@sha256:034e1e396d668bd5faea285d8ff6f7f46b483e9efb3a25b46a3c69cf129b60aa                                 "/bin/sh -c '/opt/bin"   About a minute ago   Up About a minute                       k8s_flanneld_flannel-network-sml8f_flannel-network-0d3r1_2f7f6920-b4ce-11e7-9afb-a0369f3f091a_1
654ba8be9355        quay.io/giantswarm/k8s-network-bridge@sha256:4715f04cce70ed32ab8301582b2ad6313cfe5890091ebde485521574b90e9f00                  "/bin/sh -c 'while [ "   About a minute ago   Up About a minute                       k8s_k8s-network-bridge_flannel-network-sml8f_flannel-network-0d3r1_2f7f6920-b4ce-11e7-9afb-a0369f3f091a_1
c1b773592a48        quay.io/giantswarm/flannel-network-health@sha256:56a1effd769c2f624d575b5fe1b5b6b7772bd2c5178462d8bc0a4fb31adb3052              "/flannel-network-hea"   41 minutes ago       Up 41 minutes                           k8s_flannel-network-health_flannel-network-sml8f_flannel-network-0d3r1_2f7f6920-b4ce-11e7-9afb-a0369f3f091a_0
04c78ec5c214        gcr.io/google_containers/pause-amd64:3.0                                                                                       "/pause"                 41 minutes ago       Up 41 minutes                           k8s_POD_flannel-network-sml8f_flannel-network-0d3r1_2f7f6920-b4ce-11e7-9afb-a0369f3f091a_0
```

Reply:
Good to know. Is there something like a pod scoped health check? If we have to do it this way like you did it fine. Just wondering.

Reply:
The current configuration for healthchecks should be the right one.